lib/iex/lib/iex/server.ex

```diff
@@ -47,13 +47,14 @@ defmodule IEx.Server do
   @doc """
   Returns the PID of the IEx evaluator process if it exists.
   """
-  @spec evaluator :: pid | nil
+  @spec evaluator :: {evaluator :: pid, server :: pid} | nil
   def evaluator() do
     case IEx.Server.local do
-      nil -> nil
+      nil ->
+        nil
       pid ->
         {:dictionary, dictionary} = Process.info(pid, :dictionary)
-        dictionary[:evaluator]
+        {dictionary[:evaluator], pid}
     end
```

Comment:
Can maybe be refactored as

```elixir
if pid = IEx.Server.local() do
  ...
else
  nil
end
```

but I don't have preference, just wanted to mention it.